tests/ensemble_evaluator/test_prefect_ensemble.py

```diff
@@ -1,211 +1,220 @@
+import ert_shared.ensemble_evaluator.entity.ensemble as ee
+import pytest
+import yaml
+from ert_shared.ensemble_evaluator.client import Client
 from ert_shared.ensemble_evaluator.config import EvaluatorServerConfig
-from ert_shared.ensemble_evaluator.evaluator import EnsembleEvaluator
-from ert_shared.ensemble_evaluator.prefect_ensemble.prefect_ensemble import (
-    PrefectEnsemble,
-)
-from ert_shared.ensemble_evaluator.prefect_ensemble.unix_step import UnixStep
-from ert_shared.ensemble_evaluator.prefect_ensemble.function_step import FunctionStep
-from ert_shared.ensemble_evaluator.prefect_ensemble.storage_driver import (
-    storage_driver_factory,
-)
 from ert_shared.ensemble_evaluator.entity import identifiers as ids
-from ert_shared.ensemble_evaluator.prefect_ensemble.client import Client
+from ert_shared.ensemble_evaluator.entity.unix_step import UnixTask
+from ert_shared.ensemble_evaluator.evaluator import EnsembleEvaluator
+from ert_shared.ensemble_evaluator.prefect_ensemble import PrefectEnsemble
+from prefect import Flow
+from prefect.run_configs import LocalRun
 from tests.ensemble_evaluator.conftest import _mock_ws
+from tests.utils import SOURCE_DIR, tmp
 
 
 def parse_config(path):
     conf_path = Path(path).resolve()
     with open(conf_path, "r") as f:
-        return yaml.safe_load(f)
+        config = yaml.safe_load(f)
+    for stage in config.get(ids.STAGES):
+        for step in stage.get(ids.STEPS):
+            for input_ in step.get(ids.INPUTS):
+                input_[ids.LOCATION] = Path(input_[ids.LOCATION])
+                for job in step.get(ids.JOBS):
+                    job[ids.EXECUTABLE] = (
+                        Path(job[ids.EXECUTABLE])
+                        if step.get(ids.TYPE) == ids.UNIX
+                        else job[ids.EXECUTABLE]
+                    )
+    return config
 
 
-def input_files(config, coefficients):
-    paths = {}
-    storage = storage_driver_factory(config.get(ids.STORAGE), ".")
-    file_name = "coeffs.json"
-    for iens, values in enumerate(coefficients):
-        with open(file_name, "w") as f:
-            json.dump(values, f)
-        paths[iens] = (storage.store(file_name, iens),)
-    return paths
+def input_transmitter(name, data, storage_path):
+    transmitter = ert3.data.SharedDiskRecordTransmitter(
+        name=name, storage_path=Path(storage_path)
+    )
+    asyncio.get_event_loop().run_until_complete(transmitter.transmit_data(data))
+    return {name: transmitter}
 
 
-@pytest.fixture()
-def coefficients():
-    return [{"a": a, "b": b, "c": c} for (a, b, c) in [(1, 2, 3), (4, 2, 1)]]
+def coefficient_transmitters(coefficients, storage_path):
+    transmitters = defaultdict(dict)
+    record_name = "coeffs"
+    for iens, values in enumerate(coefficients):
+        transmitters[iens] = input_transmitter(record_name, values, storage_path)
+    return dict(transmitters)
+
+
+def script_transmitters(config):
+    transmitters = defaultdict(dict)
```

Comment:
I don't understand the purpose of using a defaultdict in this nor `coefficient_transmitters` function...

Reply:
In the . But here it's used to be able to do `transmitters[iens].update(..)` without always first doing
```
if iens not in transmitters:
    transmitters[iens] = dict()
```

Reply:
Ah, yes! You are right! I did spot a few places where `defaultdict()` and then I don't really see the point. I agree that for the script transmitter it actually has a purpose.